### PR TITLE
[DOPS-908] Makefile improvements

### PR DESCRIPTION
**Pull request by phoinixgrr**
#### Summary

This introduces the following improvements: 

- [fix] Linting docker cmd alias for go-lint https://github.com/mattermost/rtcd/pull/26/commits/e58c16dbe17c36d544476ba456b21d3eee6bd3c7
- [feat] Makefile addition of docker-sign and docker-verify https://github.com/mattermost/rtcd/commit/a12541c109a5510b8537b4e05c808ac09cd8bb0a
- [feat] Makefile addition of go-fmt https://github.com/mattermost/rtcd/commit/c71a2cefbdbd77ec8639c6ab581fd0e3bdd5ba72
- [feat] Makefile build for FreeBSD😈https://github.com/mattermost/rtcd/commit/83e0f19b985ea36ead1f242133737c91c0d68331
- [feat] Makefile addition of verbosity control https://github.com/mattermost/rtcd/commit/7f2d7452ba6e7d8ad2374669315099de4c82d86f

> Setting `VERBOSE=1` will increase the verbosity. Otherwise make is mostly silent. 
> Example:

```
❯ make docker-lint
15:10:16 [ .. ] Dockerfile linting...
15:10:24 [ OK ] Dockerfile linting

❯ VERBOSE=1 make docker-lint                                                                                                                                                               
15:10:29 [ .. ] Dockerfile linting...
/usr/local/bin/docker run -i --rm -u $(id -u):$(id -g) --platform "linux/amd64" \
	"hadolint/hadolint:v2.9.2@sha256:d355bd7df747a0f124f3b5e7b21e9dafd0cb19732a276f901f0fdee243ec1f3b" \
	< ./build/Dockerfile || (echo `date +%H:%M:%S` [FAIL] && false)
15:10:35 [ OK ] Dockerfile linting
```

Review by commit is recommended. 

#### Ticket Link

Ticket: https://mattermost.atlassian.net/browse/DOPS-908

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,35 +116,66 @@ test: go-test ## to test all
...
+	cosign sign --key cosign.key $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	$(AT)rm -f cosign.key || ${FAIL}
+	@$(OK) Signing the docker image: $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}/${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-verify
+docker-verify: ## to verify the docker image
+	@$(INFO) Verifying the published docker image...
+	$(AT)echo "$${COSIGN_PUBLIC_KEY}" > cosign_public.key && \
+	$(DOCKER) run ${DOCKER_OPTS} \
+	--entrypoint '/bin/sh' \
```
[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile:152]: Yes we do, nice catch ! 
Rebased and pushed changes

File: Makefile [hunk 2]
```diff
@@ -83,6 +91,14 @@ ERR  = echo ${TIME} ${RED}[FAIL]${CNone}
 OK   = echo ${TIME} ${GREEN}[ OK ]${CNone}
 FAIL = (echo ${TIME} ${RED}[FAIL]${CNone} && false)
 
+# ====================================================================================
+# Verbosity control hack
+
+VERBOSE ?= 0
+AT_0 := @
+AT_1 :=
+AT = $(AT_$(VERBOSE))
```
[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile:100]: Yes it is!!! 
Credits https://stackoverflow.com/questions/9314322/controlling-verbosity-of-make

File: Makefile [hunk 3]
```diff
@@ -160,50 +207,56 @@ go-build/%:
...
-	$(GO) test ${GO_TEST_OPTS} ./... || ${FAIL}
+	$(AT)$(GO) test ${GO_TEST_OPTS} ./... || ${FAIL}
 	@$(OK) testing
 
 .PHONY: go-mod-check
 go-mod-check: ## to check go mod files consistency
 	@$(INFO) Checking go mod files consistency...
-	go mod tidy
-	git --no-pager diff --exit-code go.mod go.sum || \
+	$(AT)go mod tidy
```
[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile]: Nice catch

File: Makefile [hunk 4]
```diff
@@ -160,50 +207,56 @@ go-build/%:
...
+	$(AT)$(GO) get -u ./... && \
+	$(AT)$(GO) mod vendor && \
+	$(AT)$(GO) mod tidy || ${FAIL}
 	@$(OK) updating go dependencies
 
 .PHONY: go-lint
 go-lint: ## to lint go code
 	@$(INFO) App linting...
-	GOCACHE="/tmp" docker run ${DOCKER_OPTS} \
+	$(AT)GOCACHE="/tmp" docker run ${DOCKER_OPTS} \
```
[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile]: Nice catch

File: Makefile [hunk 5]
```diff
@@ -160,50 +207,56 @@ go-build/%:
...
 	-e GOCACHE="/tmp" \
 	-e GOLANGCI_LINT_CACHE="/tmp" \
 	${DOCKER_IMAGE_GOLINT} \
 	golangci-lint run ./... || ${FAIL}
 	@$(OK) App linting
 
+.PHONY: go-fmt
+go-fmt: ## to perform formatting
+	@$(INFO) App code formatting...
+	$(AT)go fmt ./... || ${FAIL}
```
[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile]: Nice catch

File: Makefile [hunk 6]
```diff
@@ -24,7 +24,15 @@ DOCKER_PASSWORD         ?= password
 ## Docker Images
 DOCKER_IMAGE_GOLINT     += "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b"
 DOCKER_IMAGE_DOCKERLINT += "hadolint/hadolint:v2.9.2@sha256:d355bd7df747a0f124f3b5e7b21e9dafd0cb19732a276f901f0fdee243ec1f3b"
+DOCKER_IMAGE_COSIGN     += "bitnami/cosign:1.8.0@sha256:8c2c61c546258fffff18b47bb82a65af6142007306b737129a7bd5429d53629a"
 
+## Cosign Variables
+# The public key
+COSIGN_PUBLIC_KEY       ?= akey
```
[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile:31]: These was designed to be used as Environment Variables. 
Would vote against using files to pass creds.

[the remark this answers is not in the document]
Comment (phoinixgrr) [reply Makefile:31]: Let's say we migrate to Github Actions, some of the functionality of the makefile, would the File variable work there?

[the remark this answers is not in the document]
Comment (pfltdv) [reply Makefile:31]: Yes you are right , github actions does not support file variables. And for Github it makes everything too complex. Resolving this thread.

**Review by pfltdv — APPROVED** [review]
:tumbler_glass: Great work.

**Review by streamer45 — APPROVED** [review]
Always improving, awesome :+1:

**Pull request merged**